### PR TITLE
refactor: remove redundant `+1` offset in `NestedSetsBehavior` class in `shiftLeftRightAttribute` calls.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/NestedSetsBehavior.php [hunk 1]
```diff
@@ -142,7 +142,7 @@ public function afterDelete(): void
         $rightValue = $this->getOwner()->getAttribute($this->rightAttribute);
 
         if ($this->operation === self::OPERATION_DELETE_WITH_CHILDREN || $this->getOwner()->isLeaf()) {
-            $this->shiftLeftRightAttribute($rightValue + 1, $leftValue - $rightValue - 1);
+            $this->shiftLeftRightAttribute($rightValue, $leftValue - $rightValue - 1);
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/NestedSetsBehavior.php:145]: Thank you for the excellent suggestion regarding tree integrity validation. You're absolutely right that adding a `validateTreeIntegrity()` method would significantly improve confidence in nested set operations and help detect off-by-one errors that current tests might miss.

However, i'd prefer to keep this PR focused on the specific refactoring of removing the functionally equivalent `+1` operations. The integrity validation you're suggesting is a valuable enhancement that deserves its own dedicated PR with proper design, implementation, and comprehensive testing.

I'll create a follow-up issue to track the implementation of:
- Tree integrity validation method checking for gaps, overlaps, and boundary correctness
- Integration points after critical operations (deletion, cross-tree movement, root promotion)
- Comprehensive test coverage for edge cases and boundary conditions

This approach allows us to:
1. Keep the current PR small and focused
2. Give proper attention to the validation feature design
3. Ensure thorough testing of the new validation capabilities

Would you agree this is a reasonable approach?
